src/main/java/org/jenkinsci/plugins/workflow/cps/Snippetizer.java

```diff
@@ -62,105 +65,235 @@
-                if (first) {
-                    first = false;
-                } else {
-                    b.append(", ");
-                }
-                Object key = entry.getKey();
-                if (key instanceof String && SourceVersion.isName((String) key)) {
-                    b.append(key);
-                } else {
-                    render(b, key);
-                }
-                b.append(": ");
-                render(b, entry.getValue());
+            Object key = entry.getKey();
+            if (key instanceof String && SourceVersion.isName((String) key)) {
+                b.append(key);
+            } else {
+                object2Groovy(b, key, true);
             }
-            b.append(']');
+            b.append(": ");
+            object2Groovy(b, entry.getValue(), true);
+        }
+    }
+
+    /**
+     * Writes out a given {@link UninstantiatedDescribable} as a function call form.
+     *
+     * @param nested
+     *      true if this object is written as a nested expression (in which case we always produce parenthesis for readability
+     */
+    private static StringBuilder ud2groovy(StringBuilder b, UninstantiatedDescribable ud, boolean blockArgument, boolean nested) {
+        if (ud.getSymbol()==null) {
+            // if there's no symbol, we need to write this as [$class:...]
+            return map2groovy(b,ud.toMap());
+        }
+
+        final Map<String, ?> args = ud.getArguments();
+
+        // if the whole argument is just one map?
+
+        // the call needs explicit parenthesis sometimes
+        //   a block argument normally requires a () around arguments, and if arguments are empty you need explicit (),
+        //   but not if both is the case!
+        final boolean needParenthesis = (blockArgument ^ args.isEmpty()) || isSingleMap(args) || nested;
+
+        b.append(ud.getSymbol());
+        b.append(needParenthesis ? '(': ' ');
+
+        if (ud.hasSoleRequiredArgument()) {
+            // lone argument optimization, which gets rid of named arguments and just write one value, like
+            // retry 5 { ... }
+            object2Groovy(b, args.values().iterator().next(), true);
         } else {
-            b.append("<object of type ").append(valueC.getCanonicalName()).append('>');
+            // usual form, which calls out argument names, like
+            // git url:'...', browser:'...'
+            mapWithoutBracket2groovy(b,args);
+        }
+
+        if (needParenthesis)
+            b.append(')');
+
+        if (blockArgument) {
+            if (!args.isEmpty())    b.append(' ');
+            b.append("{\n    // some block\n}");
+        }
+
+        return b;
+    }
+
+    /**
+     * If the sole argument is a map, its [...] bracket cannot be present.
+     *
+     * Historically we've disambiguated this by adding (...) around the function call.
+     * TODO: I claim removing both () and [] would be better.
```

Comment:
OK, well make that change and write a test running the script using that syntax to prove that `DSL` actually gets the write parameters.